### PR TITLE
SDK, reorganize mod structure for clarity

### PR DESCRIPTION
**Pull request by jul-sh**
Moves a couple of utility functions from the root level into a pub channel and utils mod. Should make the docs a lot easier to read. Also aligns the crate file structure with its pub interface for clarity.

The diff is a bit misleading, but code has only been moved between files, not written anew. 

This is technically a breaking change, but all moved functions are very new, and I don't expect teams to already be depending on them. And if they are, the compiler error message should make migration not too difficult.

### CODE REVIEW

**Comment by andrisaar** [comment]
Proposal: move all the mock attestation related stuff into `mock_attestation.rs` or something like that.

This would leave the root `lib.rs` much cleaner, and mean you'd only have to tag `mod mock_attestation` with the feature gate.

(Honestly I'd probably move all the non-mock stuff to a different file as well, keeping only the interfaces in the root lib.rs file.)

File: oak_restricted_kernel_sdk/src/lib.rs [hunk 1]
```diff
@@ -18,23 +18,18 @@
...
+pub use oak_crypto::encryptor::EncryptionKeyHandle;
+use oak_crypto::{encryptor::EncryptionKeyProvider, hpke::RecipientContext};
+use oak_dice::evidence::{Evidence, RestrictedKernelDiceData, P256_PRIVATE_KEY_SIZE};
+use oak_restricted_kernel_interface::{syscall::read, DICE_DATA_FD};
 /// Marks a function as the entrypoint to an enclave app and sets up an conviences such an
 /// allocator, logger, panic handler.
 ///
 /// This macro assumes that crates using it have declared the [`no_std`] and [`no_main`]
-/// attributes, and the [`alloc_error_handler`] unstable feature of the rust compiler.
+/// attributes, and the enabled [`alloc_error_handler`] unstable feature of the rust compiler.
```
Comment (tiziano88) [inline oak_restricted_kernel_sdk/src/lib.rs:32]:
```suggestion
/// attributes, and enabled the [`alloc_error_handler`] unstable feature of the rust compiler.
```

**Review by tiziano88 — APPROVED** [review]

**Comment by tiziano88** [comment]
> Proposal: move all the mock attestation related stuff into `mock_attestation.rs` or something like that.
> 
> This would leave the root `lib.rs` much cleaner, and mean you'd only have to tag `mod mock_attestation` with the feature gate.
> 
> (Honestly I'd probably move all the non-mock stuff to a different file as well, keeping only the interfaces in the root lib.rs file.)

+1 all the mock stuff is quite invasive at the moment, a sub module may make that tidier

File: oak_core/src/samplestore.rs [hunk 2]
```diff
@@ -14,6 +14,8 @@
 // limitations under the License.
 //
 
+//! Probabilisting sample store to compute percentiles over some measurement.
```
Comment (conradgrobler) [inline oak_core/src/samplestore.rs:17]: nit: Probabilisting -> Probabilistic, also pre-existing below

**Review by conradgrobler — APPROVED** [review]

**Review by ipetr0v — APPROVED** [review]

**Comment by jul-sh** [comment]
I've moved the attestation structs to `instance_attestation` and `mock_attestation` submodules respectively. I agree it makes the crate a lot more readable.

@tiziano88 @ipetr0v What do you both think? Imo this seems nice, but then the struct naming becomes a bit redundant. 

eg:

```
oak_restricted_kernel_sdk::mock_attestation::MockEvidenceProvider,
```

etc. Should we shorten the struct names? (imo probs yes). Should instead we keep it all in one mod? Many choices.

**Review by andrisaar — APPROVED** [review]

**Comment by ipetr0v** [comment]
> I've moved the attestation structs to `instance_attestation` and `mock_attestation` submodules respectively. I agree it makes the crate a lot more readable.
> 
> @tiziano88 @ipetr0v What do you both think? Imo this seems nice, but then the struct naming becomes a bit redundant.
> 
> eg:
> 
> ```
> oak_restricted_kernel_sdk::mock_attestation::MockEvidenceProvider,
> ```
> 
> etc. Should we shorten the struct names? (imo probs yes). Should instead we keep it all in one mod? Many choices.

I don't think we'll ever have a Group Attestation, so the instance attestation name may be confusing. So maybe just `attestation`?

Also I think `MockAttestation` also be moved to tests or testing, since it will only used for tests.

**Comment by jul-sh** [comment]
> I don't think we'll ever have a Group Attestation, so the instance attestation name may be confusing. So maybe just `attestation`?

Idk, seems possible we may want that at some point

> Also I think `MockAttestation` also be moved to tests or testing, since it will only used for tests.

like public testing module? It needs to be public, since it's exported for to be used by other crates in their tests

**Pull request merged**